clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/CacheController.groovy

```diff
@@ -36,19 +36,35 @@ class CacheController {
                                 @PathVariable String type,
                                 @RequestBody Map<String, ? extends Object> data) {
     OnDemandAgent.OnDemandType onDemandType = getOnDemandType(type);
-    def cacheStatus = onDemandCacheUpdaters.find { it.handles(onDemandType, cloudProvider) }?.handle(onDemandType, cloudProvider, data)
+
+    def onDemandCacheResult = onDemandCacheUpdaters.find {
+      it.handles(onDemandType, cloudProvider)
+    }?.handle(onDemandType, cloudProvider, data)
+
+    def cacheStatus = onDemandCacheResult?.status
     def httpStatus = (cacheStatus == OnDemandCacheUpdater.OnDemandCacheStatus.PENDING) ? HttpStatus.ACCEPTED : HttpStatus.OK
-    return new ResponseEntity(httpStatus)
+
+    return new ResponseEntity(
+      [
+        cachedIdentifiersByType: onDemandCacheResult?.cachedIdentifiersByType ?: [:]
+      ],
+      httpStatus
+    )
   }
 
   @RequestMapping(method = RequestMethod.GET, value = "/{cloudProvider}/{type}")
-  Collection<Map>  pendingOnDemands(@PathVariable String cloudProvider,
-                                    @PathVariable String type) {
+  Collection<Map> pendingOnDemands(@PathVariable String cloudProvider,
+                                   @PathVariable String type,
+                                   @RequestParam(name = "id", required = false) String id) {
```

Comment:
Any thought on this as a `RequestParam` vs `PathVariable`?

The latter would look something like `/{cloudProvider}/{type}/{id}`. I actually did this originally but figured it might be more annoying to deal with in `orca` (ie. empty collection vs. 404) and went back to it just being a filter. 

Happy to revisit.